### PR TITLE
Fix build errors caused by boundary test conditions

### PR DESCRIPTION
**Pull request by aditi-dsi**
## Description
This PR fixes the build error occurred from one of the boundary test condition in `hostedChallengesCtrl.test.js` by updating it to:
- Freeze the date and time with `jasmine.clock()` to ensure all test environments run with a fixed date and time to avoid corner condition failures.
- Then cleans up the mock clock and restores the real clock after the boundary tests are completed.

### CODE REVIEW

File: frontend/tests/controllers-test/hostedChallengeCtrl.test.js [hunk 1]
```diff
@@ -169,8 +169,15 @@ describe('Unit tests for hosted challenge controller', function () {
     
     describe('Challenge categorization with duplicate handling', function () {
         beforeEach(function () {
+            jasmine.clock().install();
+            jasmine.clock().mockDate(new Date("2025-01-01T12:00:00Z"));
```
Comment (RishabhJain2018) [inline frontend/tests/controllers-test/hostedChallengeCtrl.test.js:173]: Hey @aditi-dsi , Can we please make this a dynamic date instead of hardcoding it?

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

File: frontend/tests/controllers-test/hostedChallengeCtrl.test.js (hunk 1 above)
Comment (aditi-dsi) [reply frontend/tests/controllers-test/hostedChallengeCtrl.test.js]: @RishabhJain2018 Done.

**Review by RishabhJain2018 — APPROVED** [review]
